server/routerlicious/packages/services-telemetry/src/schema.ts

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ILumberjackSchemaValidator, ILumberjackSchemaValidationResult, SchemaProperties } from "./resources";
+
+export abstract class BaseLumberjackSchemaValidator implements ILumberjackSchemaValidator {
+    protected readonly validators = new Map<string, (propvalue: string) => boolean>();
+
+    public validate(props: Map<string, any>): ILumberjackSchemaValidationResult {
+        const validationFailedForProperties: string[] = [];
+        this.validators.forEach((validator, keyName) => {
+            // Validation should pass if the required propery is included and satisfies the requirements.
+            if (props.has(keyName) && validator(props.get(keyName))) {
+                return;
+            }
+
+            // Otherwise, validation should fail and the property name is added to the "failed" list.
+            validationFailedForProperties.push(keyName);
+        });
+
+        return {
+            validationPassed: validationFailedForProperties.length === 0,
+            validationFailedForProperties,
+        };
+    }
+
+    // Validators
+    protected readonly idValidation = (propValue: any) => {
+        return this.isUndefined(propValue)
+            || (this.isString(propValue) && propValue.length > 0);
+    };
+
+    protected readonly seqNumberValidation = (propValue: any) => {
+        return this.isUndefined(propValue)
+            || (this.isNumber(propValue) && propValue >= -1);
+    };
+
+    // Helpers
+    private isUndefined(propValue: any): propValue is undefined {
+        return propValue === undefined;
+    }
+
+    private isNumber(propValue: any): propValue is number {
+        return typeof(propValue) === "number";
+    }
+
+    private isString(propValue: any): propValue is string {
+        return typeof(propValue) === "string";
+    }
+}
+
+export class RequestSchemaValidator extends BaseLumberjackSchemaValidator {
```

Comment:
Would `DocumentSchemaValidator` or `SessionSchemaValidator` be a better than for this? I usually don't think of Kafka lambdas are request handlers.

Reply:
Good point - I was actually having trouble coming up with a name :) Updated the name to `DocumentSchemaValidator`. Thanks!